package.json

```diff
@@ -8,25 +8,18 @@
       "./demos/simple-storage/*"
     ],
     "nohoist": [
-      "**/graph-ts",
-      "**/graph-cli",
       "**/@testing-library/*",
       "**/@types/testing-library*"
     ]
   },
   "scripts": {
-    "build": "lerna run build --no-private",
-    "build:cli": "yarn workspace @web3api/cli build",
-    "build:client-js": "yarn workspace @web3api/client-js build",
-    "build:wasm-as": "yarn workspace @web3api/wasm-as build",
+    "build": "lerna run build --no-private --ignore @web3api/cli && yarn build:cli",
```

Comment:
The reason for this is the fact that there's a circular dependency between @web3api/cli and @web3api/client-js. So to fix this, we build the client first, since it only requires @web3api/cli as a devDependency

Reply:
Might be wrong but I'm pretty sure lerna prioritizes dep over devDep and warny about circular

Reply:
Ya I thought that would be the case too. It looks like it is, but the fact that we're deleting the build directory with `rimraf` before compiling with `tsc` causes the build to fail like so:
```
lerna info run Ran npm script 'build' in '@web3api/client-js' in 5.3s:
$ rimraf ./build && tsc --project tsconfig.build.json
lerna ERR! yarn run build exited 2 in '@web3api/cli'
lerna ERR! yarn run build stdout:
$ rimraf ./build && tsc
src/commands/query.ts(10,49): error TS2307: Cannot find module '@web3api/client-js' or its corresponding type declarations.
src/lib/Compiler.ts(12,59): error TS2307: Cannot find module '@web3api/client-js' or its corresponding type declarations.
src/lib/Web3APIManifest.ts(3,26): error TS2307: Cannot find module '@web3api/client-js' or its corresponding type declarations.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lerna ERR! yarn run build stderr:
error Command failed with exit code 2.
```

The use of `rimraf` before `tsc` is required so that we can remove old build artifacts that may not exist anymore. Here's an issue where this is talked about: https://github.com/microsoft/TypeScript/issues/13722

I'll look into how this can fixed. Maybe by running to yarn commands instead of using the inline `&&` to concatenate the CLI commands. I think what's happening is that lerna when it first runs takes a snapshot of all the files it needs and stores them in memory. These snapshotted file descriptors may become invalid after a deletion occurs.

Reply:
No luck :P will revisit this in the future, but for now we'll just try our best to avoid circular build dependencies.